### DISCUSSION TITLE
[#1040908] Worldedit-Esque Tools + Inventory-like Selection

### DISCUSSION THREAD

**Discussion by MinecraftMoments** [Mojang/minecraft-editor#48] [Feedback]
as a minecraft creator of sorts, i enjoy using the worldedit mod to help me build things in the java version of the game. i have searched for an alternative to worldedit for bedrock before, but no bedrock "mods" matched the quality. this is where the official editor perks my interest and excites me to the possibilities of what one can build without having to convert worlds across versions.

with that in mind, i want to see a few tools that are used in worldedit to be placed in the editor; for example, the ability to change biomes for an area and tools that change the terrain like the gravity, raise/lower, and smooth brushes.
i also suggest, for when picking the blocks the editor is using with the brush tool and the like, for the ability to open up a menu that looks similar to the inventory, just without the item blocks. this is not a worldedit feature, but i think it would make it easier for one to pick the possible blocks (or select multiple blocks and compare the textures quite easily when using tools that can place multiple types of blocks. adding that in because i know the multiple block percentage thing has been requested and this just adds onto that idea lol).

(also, not related to the discussion here but i dont wanna create a new discussion just for this comment: i really really enjoy the crosshair mode. while, yes, it is buggy in its current state bc is in beta, i do like the ability to noclip through blocks (presuming that's intentional lol) while basically being in creative mode. it makes me want that in java edition alongside worldedit :P it'd be way fun)

**Comment by Landy005** [comment]
This is partly a copy of what's mentioned here; https://github.com/Mojang/minecraft-editor/discussions/30 & https://github.com/Mojang/minecraft-editor/discussions/10 

But I do like the inventory input for selecting a block you want to fill an area with. It'd be much easier to use icons (recognisable) then the current texts!

I made a little mockup for it, to make it less vague; 
( I forgot to put a history bar at the top, just for convenience to have your last blocks used available)
![mockup_editor](https://user-images.githubusercontent.com/16018643/223941801-c21fbf1f-382c-4309-9214-765ab8551065.png)

**Reply by minecraftchloe** [reply]
Oh my goodness, YES!!! We've got mockups and prototypes in progress to improve the block palette experience, and also expand to include Editor-specific hotbars. Some dependencies we need to resolve first, but it's absolutely planned!
